### PR TITLE
Introduce lint-md and compliance-docs pipeline

### PR DESCRIPTION
**Pull request by silverwind**
Alternative to https://github.com/go-gitea/gitea/pull/24017, see discussion there.

- Add new Make target `lint-md`
- Add new Drone pipeline `compliance-docs`
- Add `*.md` to docs exclusion/inclusion
- Consistently quote `path.include` and `path.exclude` statements in YAML

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -345,6 +346,10 @@ lint-frontend: node_modules
 	npx eslint --color --max-warnings=0 --ext js,vue web_src/js build *.config.js docs/assets/js tests/e2e
 	npx stylelint --color --max-warnings=0 web_src/css
 	npx spectral lint -q -F hint $(SWAGGER_SPEC)
+	@$(MAKE) --no-print-directory lint-md
```
Comment (yardenshoham) [inline Makefile]: Can we change `lint-frontend: node_modules` to `lint-frontend: node_modules lint-md` instead?

Comment (silverwind) [reply Makefile]: Possible. It would change execution order but avoids the extra sub `make` which I guess is a plus.

Comment (silverwind) [reply Makefile]: Done

**Review by lunny — APPROVED** [review]

**Review by yardenshoham — APPROVED** [review]

**Review by a1012112796 — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
@silverwind sorry merged the pull that let the ci fail ... can we just push the fix here ... :/

**Comment by silverwind** [comment]
No problem. I guess the next docs-only PR will verify this new pipeline then.

**Comment by silverwind** [comment]
Hmm, did https://github.com/go-gitea/gitea/pull/24015 actually break the CI again? In any case, let's get this in now.

**Comment by 6543** [comment]
I'll force merge ...

**Comment by 6543** [comment]
Next doc pull **must** comply

**Comment by silverwind** [comment]
https://github.com/go-gitea/gitea/pull/24024

**Pull request merged**